jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBufferPool.java

```diff
@@ -36,23 +36,38 @@
      * Finds a {@link RetainableByteBufferPool} implementation in the given container, or wrap the given
      * {@link ByteBufferPool} with an adapter.
      * @param container the container to search for an existing memory pool.
-     * @param byteBufferPool the {@link ByteBufferPool} to wrap if no memory pool was found in the container.
-     * @return the {@link RetainableByteBufferPool} found or the wrapped one.
+     * @param byteBufferPool Use  {@link ByteBufferPool#asRetainableByteBufferPool()} to convert if no memory pool was found in the container.
+     * @return the {@link RetainableByteBufferPool} found or the converted one.
      */
     static RetainableByteBufferPool findOrAdapt(Container container, ByteBufferPool byteBufferPool)
     {
         RetainableByteBufferPool retainableByteBufferPool = container == null ? null : container.getBean(RetainableByteBufferPool.class);
-        if (retainableByteBufferPool == null)
+        return retainableByteBufferPool == null ? byteBufferPool.asRetainableByteBufferPool() : retainableByteBufferPool;
```

Comment:
We are not adding it.    I'm just supportin the use-case where a user has added their own with a specific configuration.    If they also configure the main pool to have a retained buffers, then there is duplication.

Perhaps, since this is a major release, we don't support this backwards compatibility?

In which case, this method just become a call to `asRetainableByteBufferPool` and can be inlined.   Maybe that is simplest.

Reply:
removed and inlined